service/state/rpc.go

```diff
@@ -13,35 +14,52 @@ import (
 	"github.com/celestiaorg/celestia-node/node/rpc"
 )
 
-var log = logging.Logger("state/rpc")
+const (
+	balanceEndpoint   = "/balance"
+	submitTxEndpoint  = "/submit_tx"
+	submitPFDEndpoint = "/submit_pfd"
+)
 
 var (
 	addrKey = "address"
 	txKey   = "tx"
+
+	log = logging.Logger("state/rpc")
 )
 
+// submitPFDRequest represents a request to submit a PayForData
+// transaction.
+type submitPFDRequest struct {
+	NamespaceID string `json:"namespace_id"`
+	Data        string `json:"data"`
+	GasLimit    uint64 `json:"gas_limit"`
+}
+
 func (s *Service) RegisterEndpoints(rpc *rpc.Server) {
-	rpc.RegisterHandlerFunc("/balance", s.handleBalanceRequest, http.MethodGet)
-	rpc.RegisterHandlerFunc(fmt.Sprintf("/balance/{%s}", addrKey), s.handleBalanceForAddrRequest, http.MethodGet)
-	rpc.RegisterHandlerFunc(fmt.Sprintf("/submit_tx/{%s}", txKey), s.handleSubmitTx, http.MethodPost)
+	rpc.RegisterHandlerFunc(balanceEndpoint, s.handleBalanceRequest, http.MethodGet)
+	rpc.RegisterHandlerFunc(fmt.Sprintf("%s/{%s}", balanceEndpoint, addrKey), s.handleBalanceForAddrRequest,
+		http.MethodGet)
+	rpc.RegisterHandlerFunc(fmt.Sprintf("%s/{%s}", submitTxEndpoint, txKey), s.handleSubmitTx, http.MethodPost)
+	rpc.RegisterHandlerFunc(submitPFDEndpoint, s.handleSubmitPFD, http.MethodPost)
 }
 
 func (s *Service) handleBalanceRequest(w http.ResponseWriter, r *http.Request) {
 	bal, err := s.accessor.Balance(r.Context())
 	if err != nil {
 		w.WriteHeader(http.StatusInternalServerError)
-		log.Errorw("serving /balance request", "err", err)
+		w.Write([]byte(err.Error())) //nolint:errcheck
```

Comment:
So this way, we send the raw error to the user? Good.

Reply:
We might make error handling a bit fancier eventually, but this will probably be part of PublicAPI work

Reply:
Yes, for now raw error as the user needs to see what's happening in the case of failure.